src/cli/commands/install.js

```diff
@@ -344,6 +351,9 @@ export class Install {
             pushDeps('dependencies', workspaceManifest, {hint: null, optional: false}, true);
             pushDeps('devDependencies', workspaceManifest, {hint: 'dev', optional: false}, !this.config.production);
             pushDeps('optionalDependencies', workspaceManifest, {hint: 'optional', optional: true}, true);
+            if (this.flags.includePeerDeps) {
```

Comment:
@arcanis I feel like this would change the lockfile resolution based on the flag which is not something we want.

I feel like we should drop the if block and use the following line:
```
pushDeps('peerDependencies', workspaceManifest, {hint: 'peer', optional: false}, this.flags.includePeerDeps);
```

Even then, I feel like this will be a big change in the final resolution. What do you think?

Reply:
That's a good point. Whether it's a valid point is probably going to be a plain decision by the product owner. 

Either we support this and running the command will invalidate the cache if used inconsistently (with and without the flag). The user may complain about the cache being broken, or it may not notice it at all. We could print a warning for this if that's desirable.

Or we don't support this feature. In that case we should mention a wontfix in the ticket, perhaps state the reason why, and close it.

I don't think it's worth a major version bump for making this a default behavior (see https://github.com/yarnpkg/yarn/pull/4689#issuecomment-336170821)

Reply:
@BYK I think there is a precedent for this (namely, the `--flat` option). But that's something that would be solved if peer dependencies were installed by default rather than via a flag (which I still think would make sense).